### DISCUSSION TITLE
Copilot (extras) suggestions overlapping LSP suggestions in cmp

### DISCUSSION THREAD

**Discussion by felipeolliveira** [LazyVim/LazyVim#4862] [Q&A] [answered]
Hi!

Most of the time, when I get autocomplete suggestions from the LSP, the popup is quickly replaced by Copilot's suggestions. If I want to go back to the LSP suggestions (like to import a package, for example), I have to close the popup with `<ctrl-e>` and reopen it with `<ctrl-space>` so that suggestions from both show up at the same time.

![2024-11-21_14-47](https://github.com/user-attachments/assets/c6cb5305-286b-4473-b737-1caa9bb430e3)
![2024-11-21_14-51](https://github.com/user-attachments/assets/a95b901d-3703-4277-8004-b30776df3c09)

How do I stop the copilot from overlapping the lsp suggestions?
(I can stop using copilot, it's a solution 😂 hehehehe)

I decided to start a discussion before opening an issue—maybe it's just me missing something!

---
### My specs
The LazyVim is up to date.
Nvim `v0.10.1`
My extras config in `$HOME/.config/nvim/lua/plugins/extras.lua`:
```lua
return {
  { import = "lazyvim.plugins.extras.lang.go" },
  { import = "lazyvim.plugins.extras.lang.typescript" },
  { import = "lazyvim.plugins.extras.lang.prisma" },
  { import = "lazyvim.plugins.extras.lang.python" },
  { import = "lazyvim.plugins.extras.ai.copilot" },
  { import = "lazyvim.plugins.extras.lsp.none-ls" },
  { import = "lazyvim.plugins.extras.linting.eslint" },
}
```
My all configs for nvim: [github.com/felipeolliveira/my-nvim-config](https://github.com/felipeolliveira/my-nvim-config)

Thank u guys!

**Comment by folke** [comment, marked as answer]
You're replacing the whole cmp config, so you're on your own.
Check the LazyVim code on how to do it right

**Reply by felipeolliveira** [reply]
You're absolutely right, it's my fault!

![cmp](https://github.com/user-attachments/assets/cfa7f58b-ac5a-4d80-9e0c-cc078eb7bac1)

I grabbed the `Supertab` recipe and started adding more stuff into the plugin. I didn’t realize I was overwriting a table before showing the suggestions.

Thank you @folke!
If you’d like, I can remove this discussion since it was just a slip on my part.
